### PR TITLE
Import video mentor

### PR DESCRIPTION
**Pull request by kycarr**
- [x] for all new queries/mutations etc. use id not _id (mongoose provides id)
- [x] for all new queries/mutations etc. use noun-first names (mediaUpload)
- [x] some details about how to calculate needsTransfer and making sure that logic is not duplicated

### CODE REVIEW

**Review by beatthat — CHANGES_REQUESTED** [review]
looks good!

Main asks are:

 - for all new queries/mutations etc. use `id` not `_id` (mongoose provides `id`)
 - for all new queries/mutations etc. use noun-first names (`mediaUpload`)
 - some details about how to calculate `needsTransfer` and making sure that logic is not duplicated

File: node/src/gql/mutation/api/update-media.ts [hunk 1]
```diff
@@ -53,10 +45,12 @@ export const mediaUpdate = {
     } else {
       media[idx] = args.media;
     }
+    const hasUntransferredMedia =
+      media.find((m) => m.needsTransfer) !== undefined;
```
[the remark this answers is not in the document]
Comment (kycarr) [reply node/src/gql/mutation/api/update-media.ts:49]: needsTransfer is already a boolean, what's the point of casting it as a boolean?

[the remark this answers is not in the document]
Comment (beatthat) [reply node/src/gql/mutation/api/update-media.ts:49]: ah, meant this. 

```typescript
Boolean(media.find((m) => m?.needsTransfer));
```

...the point is we generally always avoid comparing things explicitly to `undefined`

File: node/src/models/Mentor.ts [hunk 2]
```diff
@@ -16,7 +16,7 @@ import {
   PaginateQuery,
   pluginPagination,
 } from './Paginatation';
-import { Answer, Status } from './Answer';
+import { Answer, mediaNeedsTransfer, Status } from './Answer';
```
Comment (beatthat) [inline node/src/models/Mentor.ts:19]: let's move that `mediaNeedsTransfer` function to a `utils/static-urls` module (see above)

File: node/src/gql/types/answer.ts [hunk 3]
```diff
@@ -18,9 +19,13 @@ export const AnswerMediaType = new GraphQLObjectType({
   fields: {
     type: { type: GraphQLString },
     tag: { type: GraphQLString },
+    needsTransfer: { type: GraphQLBoolean },
     url: {
       type: GraphQLString,
       resolve: function (media: AnswerMedia) {
+        if (media.needsTransfer || media.url.startsWith('http')) {
```
Comment (beatthat) [inline node/src/gql/types/answer.ts:26]:
```typescript
import { toAbsoluteUrl } from 'utils/static-urls'
return toAbsoluteUrl(media.url)
```

File: node/src/models/Mentor.ts [hunk 4]
```diff
@@ -206,7 +211,14 @@ MentorSchema.statics.import = async function (
       }
     }
   }
-  for (const a of json.answers) {
+  for (const a of json.answers || []) {
+    for (const m of a.media || []) {
+      m.needsTransfer = mediaNeedsTransfer(m);
+      m.url = m.needsTransfer
```
Comment (beatthat) [inline node/src/models/Mentor.ts:217]:
```typescript
import { mediaNeedsTransfer, toRelativeUrl } from 'utils/static-urls'

m.needsTransfer = mediaNeedsTransfer(m.url);
m.url = toRelativeUrl(m.url);
```

```

**Review by beatthat — CHANGES_REQUESTED** [review]
looks really close, had one more idea for how to consolidate/encapsulate the logic about static urls, see inline

**Review by beatthat — DISMISSED** [review]

**Review by beatthat — APPROVED** [review]
looks great! once it's tested in integrate env, merge it

**Pull request merged**